agent/client/workflow_listen_for_ds_connection_tests.go

```diff
@@ -26,8 +27,22 @@ func (c *Client) startDataStoreConnectionTestListener(ctx context.Context) error
 				return
 			}
 
+			if err != nil && isConnectionError(err) {
+				err = retry.Do(func() error {
+					return c.reconnect()
+				})
+				if err == nil {
+					// everything was reconnect, so we can exist this goroutine
```

Comment:
what does this comment mean? I don't understand it :S

Reply:
When we call `reconnect`. It needs to restart the client. Which calls this function again with amother goroutine